### ISSUE TITLE
when using `waitForDomChange` on chrome 70 : `setImmediate` is not defined

### ISSUE THREAD

**Issue by brucou** [kentcdodds/dom-testing-library#184]
- `dom-testing-library` version:
3.16.1
- `react` version:
not using react
- `node` version:
v8.2.0
- `npm` (or `yarn`) version:
v5.3.0
### Relevant code or config:

```js
  waitForDomChange({ container: root })
```

### What happened:

```
Uncaught ReferenceError: setImmediate is not defined
    at onDone (dom-testing-library.js:4344)
    at MutationObserver.<anonymous> (dom-testing-library.js:4338)
```

### Reproduction:
can't reproduce in codesandbox. When I try to import {waitForDomChange} from "dom-testing-library" :  

```
(0 , _domTestingLibrary.waitForDomChange) is not a function
```

### Problem description:

That code from the library use a function which is not defined in chrome : 

```
function waitForDomChange(_temp) {
    var _ref = _temp === void 0 ? {} : _temp,
      _ref$container = _ref.container,
      container = _ref$container === void 0 ? getDocument() : _ref$container,
      _ref$timeout = _ref.timeout,
      timeout = _ref$timeout === void 0 ? 4500 : _ref$timeout,
      _ref$mutationObserver = _ref.mutationObserverOptions,
      mutationObserverOptions = _ref$mutationObserver === void 0 ? {
        subtree: true,
        childList: true,
        attributes: true,
        characterData: true
      } : _ref$mutationObserver;

    return new Promise(function (resolve, reject) {
      var timer = setTimeout(function () {
        onDone(new Error('Timed out in waitForDomChange.'), null);
      }, timeout);
      var observer = newMutationObserver(function (mutationsList) {
        onDone(null, mutationsList);
      });
      observer.observe(container, mutationObserverOptions);

      function onDone(error, result) {
        clearTimeout(timer);
        setImmediate(function () {
          return observer.disconnect();
        });

        if (error) {
          reject(error);
        } else {
          resolve(result);
        }
      }
    });
  }
```

**Comment by alexkrolick** [comment]
Very strange - checking the console for this page (github): `window.setTimeout` and `window.setInterval` are both available as globals, but `window.setImmediate` is not. It's probably polyfilled in. 

https://caniuse.com/#search=setImmediate

`setTimeout(fn, 0)` should be a fine replacement

**Comment by brucou** [comment]
yeah it is used in only 5% of the browsers...  I mocked it as you suggested, but that would be much better replaced at the source. What do you think? It is only used in two places in the code, for `waitForDomChange` and `waitForElement`.

**Comment by alexkrolick** [comment]
Yes, it should be replaced in the library. Care to make a PR?

**Comment by brucou** [comment]
eer how do I do that?

**Comment by kentcdodds** [comment]
Hi @brucou 👋

We're here to help you! Start here: makeapullrequest.com

Let us know if you need help!

**Comment by brucou** [comment]
amazing, looking forward to reading it. I always wondered how that works.

**Comment by kentcdodds** [comment]
@brucou,

Are you still interested in making this contribution?

**Comment by brucou** [comment]
xmas and new year\s eve got in the way. Still have a tough week. If there is no rush, let a week pass and if I haven't done anything about it by then, then go ahead and do the change. Would that work?

**Comment by kentcdodds** [comment]
Yep! No worries 😁

**Comment by brucou** [comment]
one question : there is a __tests__ directory with jest tests. I am going to test this feature in a real browser, so is it ok if I use Qunit? and create another directory called tests instead of __tests__? or what do you advise?

**Comment by kentcdodds** [comment]
That's a great point @brucou. I'm not sure. I think it'd probably be cool to have a test or two that run in a real browser (using Karma), but I think I'd rather use mocha for two reasons:

1. I'm personally more familiar with mocha
2. mocha is a LOT more widely used than Qunit

I don't think it's absolutely necessary to have tests that run in a real browser, but if you really would like to do this then you're welcome to do so :)

**Comment by brucou** [comment]
I used mocha in the past, I am cool with it but I have no appetite to
learn karma and the associated config though.
PR done.

On 1/4/19, Kent C. Dodds <notifications@github.com> wrote:
> That's a great point @brucou. I'm not sure. I think it'd probably be cool to
> have a test or two that run in a real browser (using Karma), but I think I'd
> rather use mocha for two reasons:
>
> 1. I'm personally more familiar with mocha
> 2. mocha is a LOT more widely used than Qunit
>
> I don't think it's absolutely necessary to have tests that run in a real
> browser, but if you really would like to do this then you're welcome to do
> so :)
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kentcdodds/dom-testing-library/issues/184#issuecomment-451353595


-- 
---

**Comment by brucou** [comment]
Forgot to thank you for the support Kent in making that pull request. That was really appreciated! and the learning material is awesome.

**Comment by kentcdodds** [comment]
Awesome! Congrats on the PR @brucou! :+1:
